plugins/@grouparoo/pipedrive/src/lib/export/exportProfile.ts

```diff
@@ -0,0 +1,128 @@
+import { ExportProfilePluginMethod } from "@grouparoo/core";
+import { connect } from "../connect";
+import {
+  PipedriveCacheData,
+  getKnownPersonFieldMap,
+} from "./destinationMappingOptions";
+import { getGroupFieldKey } from "./listMethods";
+
+export const exportProfile: ExportProfilePluginMethod = async ({
+  appId,
+  appOptions,
+  export: {
+    newProfileProperties,
+    oldProfileProperties,
+    newGroups,
+    oldGroups,
+    toDelete,
+  },
+}) => {
+  if (Object.keys(newProfileProperties).length === 0) {
+    return { success: true };
+  }
+
+  const client = await connect(appOptions);
+  const cacheData: PipedriveCacheData = { appId, appOptions };
+
+  const oldEmail = oldProfileProperties["Email"];
+  const foundId = await client.EnhancedPersonsController.findPersonIdByEmail(
+    oldEmail
+  );
+
+  if (toDelete) {
+    if (!foundId) {
+      throw new Error(`Person with email ${oldEmail} was not found to delete.`);
+    }
+
+    await client.PersonsController.deleteAPerson(foundId);
+    return { success: true };
+  }
+
+  const payload = await makePayload(
+    client,
+    cacheData,
+    oldProfileProperties,
+    newProfileProperties,
+    oldGroups,
+    newGroups
+  );
+
+  if (foundId) {
+    // Update existing Person
+    await client.EnhancedPersonsController.updateAPerson(foundId, payload);
+  } else {
+    // Create new Person
+    await client.PersonsController.addAPerson({ body: payload });
+  }
```

Comment:
assuming we tried a bit harder to understand `oldFoundId` and `newFoundId`, we would use `const foundId = newFoundId || oldFOundId` here.
You are relying a bit much overall on core handling use `old` to know someone is in the system.
Someone could have been in the system ahead of when grouparoo syncs for the first time.

Reply:
I believe it would be `oldFoundId || newFoundId` in that case, see my previous comment.